scipy/sparse/_index.py

```diff
@@ -172,18 +172,18 @@ def __setitem__(self, key, x):
             raise IndexError('number of row and column indices differ')
 
         if issparse(x):
+            if 0 in x.shape:
+                return
             if i.ndim == 1:
                 # Inner indexing, so treat them like row vectors.
                 i = i[None]
                 j = j[None]
+            x = x.tocoo(copy=True).reshape(x._shape_as_2d)
```

Comment:
Minor performance improvement:

```suggestion
            x = x.tocoo(copy=False).reshape(x._shape_as_2d, copy=True)
```